go-controller/pkg/factory/factory.go

```diff
@@ -349,7 +348,7 @@ func newQueuedInformer(oType reflect.Type, sharedInformer cache.SharedIndexInfor
 		queueWg := &sync.WaitGroup{}
 		queueWg.Add(len(adds))
 		for j := range adds {
-			adds[j] = make(chan interface{}, 1)
+			adds[j] = make(chan interface{}, 10)
```

Comment:
Not a huge fan of buffered channels. Having said that, we will probably be getting rid of this factory business soon after @dave-tucker does his magic.

Reply:
Yeah, @danwinship will have something to say about this. I originally had a larger buffer, but Dan convinced me to scale it back because really, if you can deadlock at depth 1, you can deadlock at depth 10 too. There's gotta be a way to leave it at 1.

Reply:
It should be greater than one, or else you have head-of-line blocking - two slow pods on the same queued handler block **all** pod changes from being processed.

10 seems like a good amount.

Reply:
@squeed talk to @danwinship.

Reply:
Yeah, any channel capacity besides 0 or 1 is almost certainly wrong. eg, see the recent openshift-sdn egress IP deadlock where Juan first bumped the capacity from 1 to 300 because that should have been way more than enough, but then had to rewrite it anyway because actually 300 didn't get rid of the deadlock.

Reply:
The other possibility is to rewrite this to have an array of pending events and only use the channel as a semaphore to signal that the array is non-empty

Reply:
Given that the whole point of having multiple informers is to allow parallel pod operations, why is a work queue wrong? If two pod updates come in quick succession, then it blocks all pod updates?

This isn't a synchronization point, it's a queue multiplexer. Increasing the channel size isn't intended to fix the deadlock, rather to avoid pre-multiplex blocking.

Reply:
ok, if it's not to prevent a deadlock then that could be valid